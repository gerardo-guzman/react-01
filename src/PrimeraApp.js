import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, subtitulo}) => {
    const numeros = 123;
    const objeto = {
        nombre: 'jerry',
        edad: 23
    };
    return ( 
        <>
            {/* <pre>{JSON.stringify(objeto, null, 3)}</pre> */}
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </>
    );
}
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un Subtitulo'
};

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
};

export default PrimeraApp;