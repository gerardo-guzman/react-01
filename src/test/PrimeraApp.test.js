import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import React from 'react';
import { shallow } from 'enzyme';


describe('Pruebas en PrimeraApp component', () => {
    /* test('debe de mostar el mensaje "Hola soy goku"', () => {
        const saludo = 'Hola, soy Goku';

        const { getByText } = render(<PrimeraApp saludo={saludo}/>); 
        expect(getByText(saludo)).toBeInTheDocument();
         
    }); */

    test('debe mostrar PimeraApp correctamente', () => {
        const saludo = 'Hola, Soy Goku!';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();

    });
});


