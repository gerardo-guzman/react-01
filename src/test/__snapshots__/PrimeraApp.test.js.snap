// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp component debe mostrar PimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku!
  </h1>
  <p>
    Soy un Subtitulo
  </p>
</Fragment>
`;
